client/lib/abtest/index.js

```diff
@@ -133,28 +154,34 @@ ABTest.prototype.isEligibleForAbTest = function() {
 	const clientLanguage = client.language || client.userLanguage || 'en';
 	const clientLanguagesPrimary = ( client.languages && client.languages.length ) ? client.languages[ 0 ] : 'en';
 	const localeFromSession = i18n.getLocaleSlug() || 'en';
-	const englishMatcher = /^en-?/i;
 
 	if ( ! store.enabled ) {
 		debug( '%s: Local storage is not enabled', this.experimentId );
 		return false;
 	}
 
-	if ( ! this.allowAnyLocale ) {
-		if ( isUserSignedIn() && user.get().localeSlug !== 'en' ) {
-			debug( '%s: User has a non-English locale', this.experimentId );
-			return false;
+	if ( this.localeTarget ) {
+		const localeMatcher = new RegExp( '^' + this.localeTarget + '-?', 'i' );
+		let isTargetLocale = true;
+
+		if ( isUserSignedIn() && user.get().localeSlug !== this.localeTarget ) {
+			debug( '%s: User has a %s locale', this.experimentId, this.localeTarget );
+			isTargetLocale = false;
 		}
-		if ( ! isUserSignedIn() && ! clientLanguage.match( englishMatcher ) ) {
-			debug( '%s: Logged-out user has a non-English navigator.language preference', this.experimentId );
-			return false;
+		if ( ! isUserSignedIn() && ! clientLanguage.match( localeMatcher ) ) {
+			debug( '%s: Logged-out user has a %s navigator.language preference', this.experimentId, this.localeTarget );
+			isTargetLocale = false;
 		}
-		if ( ! isUserSignedIn() && ! clientLanguagesPrimary.match( englishMatcher ) ) {
-			debug( '%s: Logged-out user has a non-English navigator.languages primary preference', this.experimentId );
-			return false;
+		if ( ! isUserSignedIn() && ! clientLanguagesPrimary.match( localeMatcher ) ) {
+			debug( '%s: Logged-out user has a %s navigator.languages primary preference', this.experimentId, this.localeTarget );
+			isTargetLocale = false;
+		}
+		if ( ! isUserSignedIn() && ! localeFromSession.match( localeMatcher ) ) {
+			debug( '%s: Logged-out user has the %s locale in session', this.experimentId, this.localeTarget );
+			isTargetLocale = false;
 		}
-		if ( ! isUserSignedIn() && ! localeFromSession.match( englishMatcher ) ) {
-			debug( '%s: Logged-out user has a non-English locale in session', this.experimentId );
+
+		if ( ( this.localeTargetExclude && isTargetLocale ) || ( ! this.localeTargetExclude && ! isTargetLocale ) ) {
```

Comment:
Last q for tonight and this one maybe a dumb one from a fuzzy brain. What's the second half of that `or` doing in human speak?  Would removing the `or` and moving the following just below accomplish the same thing or am I looking at this wrong?

```
if ( ! isTargetLocale ) {
  return false;
}
```